ecllibrary/std/Date.ecl

```diff
@@ -531,6 +531,64 @@ EXPORT INTEGER DaysBetween(Date_t from, Date_t to) := FUNCTION
     RETURN toDays - fromDays;
 END;
 
+/**
+ * Helper function.  Calculates the 1-based week number of a date, starting from
+ * a reference date.  Weeks are defined as Sunday through Saturday.
```

Comment:
Internationally there are some other representation of the week:

|System| First day of week | Used by-in|
| -- | -- | -- |
|ISO-8601 | Monday |  EU and most of other European countries, most of Asia and Oceania|
(Middle Eastern) | Saturday |  Much of the Middle East|
(North American and islamic) | Sunday | Canada, United States, India, Japan, Taiwan, Hong Kong, Macau, Israel, South Africa, most of Latin America|

Reply:
That's a good point.  Also, a specific application may locally define the start of a week differently for other reasons (e.g. business policy).  I'll make some changes to accommodate this.